lib/CodeGen/CGExpr.cpp

```diff
@@ -1689,6 +1799,24 @@ llvm::Value *CodeGenFunction::emitAbiEncode(const CallExpr *CE) {
   return Bytes;
 }
 
+ExprValuePtr CodeGenFunction::emitAbiDecode(const CallExpr *CE) {
+  // abi_encode
+  auto Arguments = CE->getArguments();
+  assert(Arguments.size() == 2);
+  auto Arg = Arguments.at(0);
+  if (auto CE = dynamic_cast<const CastExpr *>(Arg))
```

Comment:
You should not use the same name here. Because the CE means `CallExpr` in this function arguments.